cli/daemon/run/infra/infra.go

```diff
@@ -311,3 +314,76 @@ func (rm *ResourceManager) UpdateConfig(cfg *config.Runtime, md *meta.Data, dbPr
 
 	return nil
 }
+
+// SQLConfig returns the SQL server and database configuration for the given database.
+func (rm *ResourceManager) SQLConfig(db *meta.SQLDatabase) (config.SQLServer, config.SQLDatabase, error) {
```

Comment:
These changes will make more sense when you read `runtime_config.go` so I suggest you come back to this file after reading the changes to `proc_groups.go` and `runtime_config.go`